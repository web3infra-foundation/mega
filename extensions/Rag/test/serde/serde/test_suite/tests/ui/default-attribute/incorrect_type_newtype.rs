// Tests that type error points to the path in attribute

use serde_derive::Deserialize;

#[derive(Deserialize)]
#[serde(default = "main")]
struct Newtype(#[serde(default = "main")] u8);

fn main() {}
