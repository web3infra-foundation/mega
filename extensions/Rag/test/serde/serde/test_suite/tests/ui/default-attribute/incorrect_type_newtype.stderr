error[E0308]: mismatched types
 --> tests/ui/default-attribute/incorrect_type_newtype.rs:6:19
  |
6 | #[serde(default = "main")]
  |                   ^^^^^^
  |                   |
  |                   expected `Newtype`, found `()`
  |                   expected due to this

error[E0308]: `match` arms have incompatible types
 --> tests/ui/default-attribute/incorrect_type_newtype.rs:7:34
  |
5 | #[derive(Deserialize)]
  |          -----------
  |          |
  |          this is found to be of type `u8`
  |          `match` arms have incompatible types
6 | #[serde(default = "main")]
7 | struct Newtype(#[serde(default = "main")] u8);
  |                                  ^^^^^^ expected `u8`, found `()`

error[E0308]: mismatched types
 --> tests/ui/default-attribute/incorrect_type_newtype.rs:6:19
  |
6 | #[serde(default = "main")]
  |                   ^^^^^^ expected `Newtype`, found `()`
7 | struct Newtype(#[serde(default = "main")] u8);
  |        ------- expected due to this

error[E0308]: mismatched types
 --> tests/ui/default-attribute/incorrect_type_newtype.rs:7:34
  |
5 | #[derive(Deserialize)]
  |          ----------- expected due to the type of this binding
6 | #[serde(default = "main")]
7 | struct Newtype(#[serde(default = "main")] u8);
  |                                  ^^^^^^ expected `u8`, found `()`
